snaxc/dialects/phs.py

```diff
@@ -0,0 +1,333 @@
+from collections.abc import Iterator, Sequence
+
+from xdsl.dialects.arith import FloatingPointLikeBinaryOperation
+from xdsl.dialects.builtin import ArrayAttr, DictionaryAttr, Float32Type, FunctionType, IndexType, StringAttr
+from xdsl.dialects.func import FuncOpCallableInterface
+from xdsl.dialects.utils import AbstractYieldOperation
+from xdsl.ir import Attribute, Block, BlockArgument, Dialect, Region, SSAValue
+from xdsl.irdl import (
+    IRDLOperation,
+    Operation,
+    irdl_op_definition,
+    lazy_traits_def,
+    operand_def,
+    opt_prop_def,
+    prop_def,
+    region_def,
+    result_def,
+    traits_def,
+    var_region_def,
+)
+from xdsl.parser import SymbolRefAttr
+from xdsl.traits import HasAncestor, HasParent, IsolatedFromAbove, IsTerminator, Pure, SymbolOpInterface, SymbolTable
+from xdsl.utils.exceptions import VerifyException
+
+
+@irdl_op_definition
+class YieldOp(AbstractYieldOperation[Attribute]):
+    """
+    Terminator operation for phs operations
+    """
+
+    name = "phs.yield"
+
+    traits = lazy_traits_def(
+        lambda: (
+            IsTerminator(),
+            HasAncestor(AbstractPEOperation),
+            Pure(),
+        )
+    )
+
+    @property
+    def data_operands(self) -> Sequence[SSAValue]:
+        """
+        Return every operand except for switches.
+        Since there are no switches here, it should return all operands
+        """
+        return self._operands
+
+
+@irdl_op_definition
+class AbstractPEOperation(IRDLOperation):
```

Comment:
```suggestion
class AbstractPEOp(IRDLOperation):
```
we are not very consistent in this, but it might be good to follow xdsl guidelines

Reply:
What is abstract about this? Why not just PEOperation?

Reply:
Agree with consistency thing, will adapt! Thanks!
This is abstract in the sense that it is not an RTL description of a PE, but I agree that the current naming it conflates with the `append_to_abstract_graph` which is a different kind of abstract. Will adapt to your suggestion ;)